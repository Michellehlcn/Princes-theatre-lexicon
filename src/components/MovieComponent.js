import React from "react";
import styled from "styled-components";
import CurrencyFormat from "react-currency-format";

const MovieContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 10px;
  width: 280px;
  box-shadow: 0 3px 10px 0 #aaa;
  cursor: pointer;
`;
const CoverImage = styled.img`
  height: 362px;
`;
const MovieName = styled.span`
  font-size: 18px;
  font-weight: 600;
  color: black;
  margin: 15px 0;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`;
const InfoColumn = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-end;
`;
const MovieInfo = styled.span`
  font-size: 18x;
  font-weight: 500;
  color: black
  text-transform: capitalize;
  
`;

const MovieComponent = (props) => {
  const { Poster, Price, Title, cinemaWorldPrice } = props.movie;

  return (
    <MovieContainer data-testid="movieComponent-1">
      <CoverImage src={Poster} alt="" />
      <MovieName>{Title}</MovieName>

      <InfoColumn>
        <MovieInfo>
          FilmWorld: &nbsp;
          <span
            style={
              Price < cinemaWorldPrice ? { color: "green" } : { color: "red" }
            }
          >
            <CurrencyFormat
              value={Price}
              displayType={"text"}
              prefix={"$"}
              decimalScale={2}
              fixedDecimalScale
            />
          </span>
        </MovieInfo>
        <MovieInfo>
          CinemaWorld: &nbsp;
          <span
            style={
              Price > cinemaWorldPrice ? { color: "green" } : { color: "red" }
            }
          >
            <CurrencyFormat
              value={cinemaWorldPrice}
              displayType={"text"}
              prefix={"$"}
              decimalScale={2}
              fixedDecimalScale
            />
          </span>
        </MovieInfo>
      </InfoColumn>
    </MovieContainer>
  );
};

export default MovieComponent;
