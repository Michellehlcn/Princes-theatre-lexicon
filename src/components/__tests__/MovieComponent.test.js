import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import MovieComponent from "../MovieComponent";
import { Poster, Price, Title, cinemaWorldPrice } from "../MovieComponent";

afterEach(() => {
  cleanup();
});

test("should render filmWorld or cinemaWorld", () => {
  const movie = { Poster, Price, Title, cinemaWorldPrice };
  render(<MovieComponent movie={movie} />);
  const movieComponentElement = screen.getByTestId("movieComponent-1");
  expect(movieComponentElement).toBeInTheDocument();
  expect(movieComponentElement).toHaveTextContent("FilmWorld:CinemaWorld:");
  expect(movieComponentElement).toContainHTML("<span>");
});

test("matches snapshot", () => {
  const movie = { Poster, Price, Title, cinemaWorldPrice };
  const tree = renderer.create(<MovieComponent movie={movie} />).toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot();
});
