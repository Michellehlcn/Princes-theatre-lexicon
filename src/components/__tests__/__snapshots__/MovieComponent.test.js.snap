// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bczRLJ dceNsr"
  data-testid="movieComponent-1"
>
  <img
    alt=""
    className="sc-gsnTZi copwBR"
  />
  <span
    className="sc-dkzDqf gPDoDE"
  />
  <div
    className="sc-hKMtZM fddidQ"
  >
    <span
      className="sc-eCYdqJ hqgjJW"
    >
      FilmWorld:
      <span
        style={null}
      />
    </span>
    <span
      className="sc-eCYdqJ hqgjJW"
    >
      CinemaWorld:
      <span
        style={null}
      />
    </span>
  </div>
</div>
`;
